table/meta_blocks.cc

```diff
@@ -232,6 +233,7 @@ bool NotifyCollectTableCollectorsOnFinish(
       }
     }
   }
+  builder->Add(user_collected_properties);
```

Comment:
We should probably guard this under s.ok() or all_succeeded or something similar cuz the existing though reductant `builder->Add(user_collected_properties);` always happen in s.ok()

Reply:
I added an intermediate `UserCollectedProperties` for this. So that we don't re-add existing property and also only add a property when its corresponding collector's `Finish` status is OK.

Reply:
With this intermediate properties, I added some assertions for when different collectors are adding property of the same name, and to protect from malicious collector that resets the properties collected from other collectors like the revised `SstFileWriterCollector` in external_sst_file_test was doing. These behaviors were silently passing previously, which were not desirable.